app/cdap/components/RecipeManagement/ApplyRecipeView.tsx

```diff
@@ -0,0 +1,219 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import T from 'i18n-react';
+import RecipeList from 'components/RecipeList';
+import { IRecipe, SortBy, SortOrder } from 'components/RecipeList/types';
+import Box from '@material-ui/core/Box';
+import { applyRecipe } from 'components/RecipeManagement/helper';
+import { directiveRequestBodyCreator } from 'components/DataPrep/helper';
+import useSnackbar from 'components/Snackbar/useSnackbar';
+import Snackbar from 'components/Snackbar';
+import DrawerWidget from 'components/common/DrawerWidget';
+import PrimaryContainedButton from 'components/shared/Buttons/PrimaryContainedButton';
+import styled from 'styled-components';
+import { format, TYPES } from 'services/DataFormatter';
+
+const PREFIX = 'features.WranglerNewUI.Recipe';
+
+interface IApplyRecipeViewProps {
+  isOpenView: boolean;
+  workspaceId: string;
+  directives: string[];
+}
+
+export const ApplyRecipeView = ({ isOpenView, workspaceId, directives }: IApplyRecipeViewProps) => {
+  const [isOpen, setIsOpen] = useState(isOpenView);
+  const [selectedRecipe, setSelectedRecipe] = useState(null);
+  const [snackbarState, setSnackbar] = useSnackbar();
+
+  let timeoutId;
+
+  const handleSelectRecipe = (selectedObject: IRecipe) => {
+    setSelectedRecipe(selectedObject);
+  };
+
+  const applySelectedRecipe = (recipeId) => {
+    const requestBody = directiveRequestBodyCreator(directives);
+    applyRecipe(recipeId, workspaceId, requestBody, applyResponseHandler, applyErrorHandler);
+  };
+
+  const applyResponseHandler = (res) => {
+    setIsOpen(false);
+    setSelectedRecipe(null);
+    setSnackbar({
+      open: true,
+      isSuccess: true,
+      message: T.translate(`${PREFIX}.applyRecipeSuccess`).toString(),
+    });
+  };
+
+  const applyErrorHandler = (err) => {
+    setIsOpen(false);
+    setSelectedRecipe(null);
+    const errorMessage = err.message;
+    setSnackbar({
+      open: true,
+      isSuccess: false,
+      message: T.translate(`${PREFIX}.applyRecipeFailed`, {
+        errorDetails: errorMessage,
+      }).toString(),
+    });
+  };
+
+  const closeSnackBar = () => {
+    setSnackbar(() => ({
+      open: false,
+    }));
+  };
+
+  useEffect(() => {
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+    }
+    if (snackbarState.open && snackbarState.isSuccess) {
+      timeoutId = setTimeout(closeSnackBar, 5000);
+    }
+  }, [snackbarState.open]);
+
+  const renderRecipeList = () => {
+    return (
+      <>
+        <h6>{T.translate(`${PREFIX}.selectRecipeToApply`)}</h6>
+        <RecipeList
+          isOpen={isOpen}
+          showAllColumns={false}
+          showActions={false}
+          selectHandler={handleSelectRecipe}
+          sortBy={SortBy.UPDATED}
+          sortOrder={SortOrder.DESCENDING}
+          pageSize={6}
```

Comment:
probably can put 6 into some constant, so in the future when people want to change it they can easily find it

Reply:
yes, Done